test/test_esapi.py

```diff
@@ -3,9 +3,81 @@
 SPDX-License-Identifier: BSD-2
 """
 
-import unittest
+import binascii
+
+from tpm2_pytss._libtpm2_pytss import ffi, lib
+from tpm2_pytss.TCTI import TCTI
+from tpm2_pytss.TSS2_Exception import TSS2_Exception
+from tpm2_pytss.types import (
+    ESYS_TR,
+    TPM2_ALG,
+    TPM2_CAP,
+    TPM2_CC,
+    TPM2_CLOCK,
+    TPM2_ECC,
+    TPM2_ECC_CURVE,
+    TPM2_GENERATED_VALUE,
+    TPM2_HC,
+    TPM2_NT,
+    TPM2_RC,
+    TPM2_RH,
+    TPM2_SE,
+    TPM2_ST,
+    TPM2_SU,
+    TPM2B_ATTEST,
+    TPM2B_AUTH,
+    TPM2B_CREATION_DATA,
+    TPM2B_DATA,
+    TPM2B_DIGEST,
+    TPM2B_ECC_PARAMETER,
+    TPM2B_ECC_POINT,
+    TPM2B_ENCRYPTED_SECRET,
+    TPM2B_EVENT,
+    TPM2B_ID_OBJECT,
+    TPM2B_IV,
+    TPM2B_MAX_BUFFER,
+    TPM2B_NAME,
+    TPM2B_NONCE,
+    TPM2B_NV_PUBLIC,
+    TPM2B_PRIVATE,
+    TPM2B_PUBLIC,
+    TPM2B_PUBLIC_KEY_RSA,
+    TPM2B_SENSITIVE,
+    TPM2B_SENSITIVE_CREATE,
+    TPM2B_SENSITIVE_DATA,
+    TPM2B_SYM_KEY,
+    TPM2B_TEMPLATE,
+    TPM2B_TIMEOUT,
+    TPMA_NV,
+    TPMA_OBJECT,
+    TPMA_SESSION,
+    TPML_AC_CAPABILITIES,
+    TPML_ALG,
+    TPML_CC,
+    TPML_DIGEST_VALUES,
+    TPML_PCR_SELECTION,
+    TPMS_ALGORITHM_DETAIL_ECC,
+    TPMS_ATTEST,
+    TPMS_ECC_POINT,
+    TPMS_NV_PUBLIC,
+    TPMS_SENSITIVE_CREATE,
+    TPMT_HA,
+    TPMT_PUBLIC,
+    TPMT_PUBLIC_PARMS,
+    TPMT_RSA_DECRYPT,
+    TPMT_SIG_SCHEME,
+    TPMT_SIGNATURE,
+    TPMT_SYM_DEF,
+    TPMT_SYM_DEF_OBJECT,
+    TPMT_TK_AUTH,
+    TPMT_TK_CREATION,
+    TPMT_TK_HASHCHECK,
+    TPMT_TK_VERIFIED,
+    TPMU_HA,
+    TPMU_SYM_KEY_BITS,
+    TPMU_SYM_MODE,
+)
```

Comment:
Is this really better than `import *`. Why not just use the qualified names?
```python3
from tpm2_pytss import TCTI as TCTI, TSS2_Exception as TSS2_Exception, types
```

Reply:
I find using qualified names unnecessary long, but I do not have a strong opinion about this. Having `from tpm2_pytss.types import TPMU_SYM_KEY_BITS` is the default setting e.g. in `black`, `isort` and pycharm. It's just the way it's commonly done because it is non-ambiguous and reasonably short, I guess.

If you want to do it like `import tpm2_pytss.types` and `types.TPMU_SYM_KEY_BITS` every time, instead, that would be fine by me.

Reply:
> I find using qualified names unnecessary long, but I do not have a strong opinion about this. Having `from tpm2_pytss.types import TPMU_SYM_KEY_BITS` is the default setting e.g. in `black`, `isort` and pycharm. It's just the way it's commonly done because it is non-ambiguous and reasonably short, I guess.

Those are fine, and since they don't have a zillion classes, aren't overly terrible to deal with.

> 
> If you want to do it like `import tpm2_pytss.types` and `types.TPMU_SYM_KEY_BITS` every time, instead, that would be fine by me.

I'd rather that than an import list that's the majority of the file.

Reply:
Ok, as it turns out I made a promise too fast: [black cannot format multiple imports per line](https://github.com/psf/black/issues/204).

To sum this up: either it's a long import list in the tests or it is a wildcard import. I prefer the former because it is unambiguous and best practice.

If you want, I can re-add the wildcard imports (but keep isort for import formatting, i.e. alphabetic sorting etc.).

Reply:
> Ok, as it turns out I made a promise too fast: [black cannot format multiple imports per line](https://github.com/psf/black/issues/204).
> 
> To sum this up: either it's a long import list in the tests or it is a wildcard import. I prefer the former because it is unambiguous and best practice.

Yeah but sometimes it just doesn't make sense in certain situations. I think this is one of those situations. Perhaps a rule or guideline here where if the import list is over 10 things long or something like that....

Generally the reason you avoid wildcard imports are things like [this](https://rules.sonarsource.com/python/RSPEC-2208):
```
It could lead to conflicts between names defined locally and the ones imported.
It reduces code readability as developers will have a hard time knowing where names come from.
It clutters the local namespace, which makes debugging more difficult.
```

For item 1, the TPM isn't really usable without the types, you need em all.
For item 2, a 100 line import I think messes with readability more, and with IDE's the resolution of the class name is easily understood.
For item 3, yep it does, but see 1.

> 
> However, if you want, I can re-add the wildcard imports (but keep isort for import formatting, i.e. alphabetic sorting etc.).

Please do that. We can revisit explicit import lists later.